keyboards/kbdfans/kbdpad/mk1/mk1.c

```diff
@@ -13,76 +13,53 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#include "i2c_master.h"
-#include "quantum.h"
+
+#include "mk1.h"
 
 #ifdef RGBLIGHT_ENABLE
+
+#include <string.h>
+#include "i2c_master.h"
 #include "rgblight.h"
+
 extern rgblight_config_t rgblight_config;
 
+// custom RGB driver
 void rgblight_set(void) {
     if (!rgblight_config.enable) {
-        for (uint8_t i = 0; i < RGBLED_NUM; i++) {
-            led[i].r = 0;
-            led[i].g = 0;
-            led[i].b = 0;
-        }
+        memset(led, 0, 3 * RGBLED_NUM);
     }
 
-    i2c_init();
     i2c_transmit(0xb0, (uint8_t*)led, 3 * RGBLED_NUM, 100);
 }
-#endif
 
-void matrix_init_kb(void) {
-#ifdef RGBLIGHT_ENABLE
-    if (rgblight_config.enable) {
-        i2c_init();
+bool rgb_init = false;
+
+void matrix_scan_kb(void) {
+    // if LEDs were previously on before poweroff, turn them back on
+    if (rgb_init == false && rgblight_config.enable) {
         i2c_transmit(0xb0, (uint8_t*)led, 3 * RGBLED_NUM, 100);
+        rgb_init = true;
     }
-#endif
-    // call user level keymaps, if any
-    matrix_init_user();
-}
 
-void matrix_scan_kb(void) {
-#ifdef RGBLIGHT_ENABLE
     rgblight_task();
-#endif
     matrix_scan_user();
-    /* Nothing else for now. */
 }
 
-__attribute__ ((weak))
-void matrix_scan_user(void) {
-}
+#endif
 
-void backlight_init_ports(void) {
-    // initialize pins D0, D1, D4 and D6 as output
-    setPinOutput(D0);
-    setPinOutput(D1);
-    setPinOutput(D4);
-    setPinOutput(D6);
```

Comment:
I assume these two pins are not actually connected to anything, and this was just an artifact of copy+paste?

Reply:
Yes, correct. D4 is the backlight PIN for all keys except numlock. D0 is the numlock LED. D1 and D6 seem unused... looking online, they seem be used for capslock and scrolllock LEDs in other ps2avrGB PCBs.